contracts/RollupPoB.sol

```diff
@@ -0,0 +1,331 @@
+pragma solidity ^0.6.1;
+import './RollupInterface.sol';
+import './lib/RollupPoBHelpers.sol';
+
+contract RollupPoB is RollupPoBHelpers{
+
+    // Rollup smart contract address
+    RollupInterface rollupInterface;
+
+    // First block where the first era begins
+    uint public genesisBlock;
+
+    // Maximum rollup transactions: either off-chain or on-chain transactions
+    uint public MAX_TX;
+
+    // Input snark definition
+    uint256 constant offChainHashInput = 3;
+
+    // Defines slot/era block duration
+    uint constant public DELAY_GENESIS = 1000;
+    uint32 constant public BLOCKS_PER_SLOT = 100;
+    uint32 constant public SLOTS_PER_ERA = 20;
+    uint constant public SLOT_DEADLINE = 20;
+
+    // Burn Address
+    address payable burn = address(0x00);
+
+    // Minimum bid to enter the auction
+    uint public constant MIN_BID = 1 ether;
+
+    // Minimum next Bid
+    uint minNumSlots = 2;
+
+    // % of bid
+    uint percentBonus = 10;
+    uint percentNextBid = 30;
+
+    // Defines operator structure
+    struct Operator {
+        address payable beneficiaryAddress;
+        address forgerAddress;
+        address withdrawAddress;
+        address bonusAddress;
+    }
+
+    // Defines bid structure
+    struct Bid {
+        uint amount;
+        bool initialized;
+    }
+
+    // Defines information of slot
+    struct InfoSlot {
+        // Indicates if at least one batch has been forged on an slot
+        bool fullFilled;
+        // current price of slot
+        uint slotPrice;
+        // accumulated bonus
+        uint accumulatedBonus;
+    }
+
+    // Mappings
+    // mapping to control bonus amount by address
+    mapping(address => uint) public bonusBalance;
+    // mapping to control withdraw bid by address
+    mapping(address => uint) public withdrawBid;
+    // mapping to control winner by slot
+    mapping(uint => Operator) public slotWinner;
+    // mapping to control bid by slot
+    mapping(uint => Bid) public slotBid;
+    // mapping to control information of slot
+    mapping(uint => InfoSlot) public infoSlot;
+
+    /**
+     * @dev Event called when an operator commits data before forging it
+     */
+    event dataCommitted(uint256 hashOffChain);
+
+    /**
+     * @dev Event called when an operator beat the bestBid of the ongoing auction
+     */
+    event newBestBid(uint32 slot, uint256 amount, uint256 price, address operator);
+
+    /**
+     * @dev RollupPoB constructor
+     * Set first block where the era will begin
```

Comment:
Comment does not apply to this contract.
Maybe it should be: "Set first block when the first slot begin"